mcs/class/corlib/System.Reflection.Emit/ModuleBuilder.cs

```diff
@@ -669,7 +669,43 @@ internal int GetToken (string str)
 			return getToken (this, member, true);
 		}
 
+		static int typespec_tokengen =  0x1bffffff;
+		static int memberref_tokengen =  0x0affffff;
+		Dictionary<MemberInfo, int> inst_tokens = new Dictionary<MemberInfo, int> ();
+
+		//
+		// Assign a pseudo token to the various TypeBuilderInst objects, so the runtime
+		// doesn't have to deal with them.
+		// For Save assemblies, the tokens will be fixed up later during Save ().
+		// For Run assemblies, the tokens will not be fixed up, so the runtime will
+		// still encounter these objects, it will resolve them by calling their
+		// RuntimeResolve () methods.
+		//
+		int GetPseudoToken (MemberInfo member) {
+			int token;
+
+			if (inst_tokens.TryGetValue (member, out token))
+				return token;
+			// Count backwards to avoid collisions with the tokens
+			// allocated by the runtime
+			if (member is MonoGenericClass)
```

Comment:
Subtypes of SymbolType. When you use MakeXXXType on a TypeBuilder.

Test what happens in your patchset if you use a MakeArrayType on a tb.